src/python/gem5/resources/client.py

```diff
@@ -163,7 +169,218 @@ def get_resource_json_obj(
                          current build. If ``None``, filtering based on compatibility
                          is not performed.
     """
+    _get_clientwrapper()
+    if resource_version:
+        resource_info = [
+            {
+                "id": resource_id,
+                "resource_version": resource_version,
+                "gem5_version": gem5_version,
+            }
+        ]
+    else:
+        resource_info = [{"id": resource_id, "gem5_version": gem5_version}]
+
+    # We will return a list when we refactor ontain_resources to handle multiple
+    # resources
+    return _get_resource_json_obj_from_client(resource_info, clients)[0]
+
+
+def get_multiple_resource_json_obj(
+    resource_info: List[Dict[str, str]],
```

Comment:
I don't like how much we're passing around `Dict[str,str]`. Why don't we create a class to hold the Query information?:

```py
class ClientQuery:
    def __init__(self, resource_id: str, version: str, gem5_version: Optional[str] = None):
       self._id = resource_id
       self._version = version
       self._gem5_version = gem5_version

    def get_id(self) -> str
        return self._id

    def get_version(self) -> str
        return self._version

    def get_gem5_version(self) -> Optional[str]
        return self._gem5_version

    def sortable_tuple(self) -> Tuple:
        to_return = (self.get_id().lower(),)
        for val in self.get_version().split("."):
            if val.isdigit():
                to_return += (int(val),)
            else:
                to_return += (str(val).lower(),)
        return to_return
```

Reply:
If we are creating this class, cant we just make the class encompass the query itself.
```
class ClientQuery:
    def __inti__(self, query_dicts: List[Dict[str,str]]):
        self.query = query_dicts

    def add_resource_to_query(self, resource_id:str, gem5_version:str,  resoruce_version:str):
        self.query.append(
            "id":resource_id, "gem5_version": gem5_version, "resource_version":resoruce_version)
```

Essentially just encapsulate the list into a class, we can also have a function that returns the query in MongoDB format if we want as well.